platform/common/core/generic/committer/finality.go

```diff
@@ -97,9 +99,13 @@ func (c *FinalityManager[V]) runEventQueue(context context.Context) {
 	for {
 		select {
 		case <-context.Done():
+			c.logger.Debugf("closing event queue")
 			return
 		case event := <-c.eventQueue:
 			c.listenerManager.InvokeListeners(event)
+			c.mutex.Lock()
```

Comment:
this makes me think about handling panics. if InvokeListeners panics, then we don't dispatch events anymore.